remove-webhook/remove-webhook.sh

```diff
@@ -1,24 +1,81 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-if [ -n "$DEBUG" ]
-then
-    set -x
+usage() {
+  cat << EOF
+usage: bash remove-webhook.sh [--insecure-skip-tls-verify]
+
+This script adjusts the rancher-webhook chart release in all clusters managed by Rancher (excluding the local cluster).
+Depending on the version of Rancher, it either deletes the downstream webhook release, adjusts its version and restarts, or does nothing.
+Requires kubectl and helm to be installed and available on \$PATH.
+
+RANCHER_URL without a trailing slash must be set with the server URL of Rancher.
+RANCHER_TOKEN must be set with an admin token generated with no scope.
+To ignore TLS verification, set --insecure-skip-tls-verify.
+
+Users also need to ensure they have the rancher-charts repo in the local Helm index.
+EOF
+}
+
+if [ "$1" == "-h" ]; then
+  usage
+  exit 0
 fi
 
-usage() {
-    echo "./remove-webhook.sh [--insecure-skip-tls-verify]"
-    echo "Remove the webhook chart in all clusters managed by rancher (excluding the local cluster)"
-    echo "Requires kubectl and helm to be installed and available on \$PATH"
-    echo "--insecure-skip-tls-verify can be set to configure the script to ignore tls verification"
-    echo "RANCHER_TOKEN must be set with an admin token generated with no scope"
-    echo "RANCHER_URL must be set with the url of rancher (no trailing /) - should be the server URL"
+delete_webhook() {
+  cluster="$1"
+  current_chart=$(helm list -n cattle-system -l name=rancher-webhook | tail -1 | cut -f 6)
+  echo "Deleting $current_chart from cluster $cluster."
+  helm uninstall rancher-webhook -n cattle-system
+}
+
+replace_webhook() {
+  cluster="$1"
+  new_version="$2"
+
+  echo "Updating the agent to make it remember the min version $new_version of rancher-webhook, so that it can deploy it when needed in the future in cluster $cluster."
+  kubectl set env -n cattle-system deployment/cattle-cluster-agent CATTLE_RANCHER_WEBHOOK_MIN_VERSION="$new_version"
+
+  helm get values -n cattle-system rancher-webhook > current_values.yaml
```

Comment:
I don't think that this works quite how we want here. When I run this on my local, I see a `USER-SUPPLIED-VALUES:` line at the top. I think that you will likely need to filter this value out, or risk mangling the values that are being supplied to the upgrade command.

Reply:
This is what happens when I capture the output with that line, edit one of the values, and upgrade with the new values pointing to the file: the upgrade succeeds, the updated value is in place. When I get the values again, this is what's returned:
```USER-SUPPLIED VALUES:
USER-SUPPLIED VALUES: null
[values follow]```

The same happens even if I re-upgrade five times. The two extra lines don't multiply...

Reply:
Right - I'm concerned that you are adding an additional value to the values of the chart, which you should not be doing. I'm also concerned that the `USER-SUPPLIED VALUES` may be causing other values to be parsed incorrectly.

Reply:
Makes sense. Do you also think we should write out all values? `helm get values --all `? That includes all computed values.

Reply:
No - I think if you use all values you will keep the current image (i.e. for the one that is currently installed). I think that's why you saw --re-use values not working during testing was that it was doing something similar to that (though that's just a guess).